tests/PendingTenantsTest.php

```diff
@@ -0,0 +1,204 @@
+use Stancl\Tenancy\Events\CreatingPendingTenant;
+use Stancl\Tenancy\Events\PendingTenantCreated;
+use Stancl\Tenancy\Events\PendingTenantPulled;
+use Stancl\Tenancy\Events\PullingPendingTenant;
+use Stancl\Tenancy\Tests\Etc\Tenant;
+
+test('tenants are correctly identified as pending', function (){
+    Tenant::createPending();
+
+    expect(Tenant::onlyPending()->count())->toBe(1);
+
+    Tenant::onlyPending()->first()->update([
+        'pending_since' => null
+    ]);
+
+    expect(Tenant::onlyPending()->count())->toBe(0);
+});
+
+test('pending trait adds query scopes', function () {
+    Tenant::createPending();
+    Tenant::create();
+    Tenant::create();
+
+    expect(Tenant::onlyPending()->count())->toBe(1)
+        ->and(Tenant::withPending(true)->count())->toBe(3)
+        ->and(Tenant::withPending(false)->count())->toBe(2)
+        ->and(Tenant::withoutPending()->count())->toBe(2);
+
+});
+
+test('pending tenants can be created and deleted using commands', function () {
+    config(['tenancy.pending.count' => 4]);
+
+    Artisan::call(CreatePendingTenants::class);
+
+    expect(Tenant::onlyPending()->count())->toBe(4);
+
+    Artisan::call(ClearPendingTenants::class);
+
+    expect(Tenant::onlyPending()->count())->toBe(0);
+});
+
+test('CreatePendingTenants command can have an older than constraint', function () {
+    config(['tenancy.pending.count' => 2]);
+
+    Artisan::call(CreatePendingTenants::class);
+
+    tenancy()->model()->query()->onlyPending()->first()->update([
+        'pending_since' => now()->subDays(5)->timestamp
+    ]);
+
+    Artisan::call('tenants:pending-clear --older-than-days=2');
+
+    expect(Tenant::onlyPending()->count())->toBe(1);
+});
+
+test('CreatePendingTenants command cannot run with both time constraints', function () {
+    pest()->artisan('tenants:pending-clear --older-than-days=2 --older-than-hours=2')
+        ->assertFailed();
+});
+
+test('CreatePendingTenants commands all option overrides any config constraints', function () {
+    Tenant::createPending();
+    Tenant::createPending();
+
+    tenancy()->model()->query()->onlyPending()->first()->update([
+        'pending_since' => now()->subDays(10)
+    ]);
+
+    config(['tenancy.pending.older_than_days' => 4]);
+
+    Artisan::call(ClearPendingTenants::class, [
+        '--all' => true
+    ]);
+
+    expect(Tenant::onlyPending()->count())->toBe(0);
+});
+
+test('tenancy can check for pending tenants', function () {
+    Tenant::query()->delete();
+
+    expect(Tenant::onlyPending()->exists())->toBeFalse();
+
+    Tenant::createPending();
+
+    expect(Tenant::onlyPending()->exists())->toBeTrue();
+});
+
+test('tenancy can pull a pending tenant', function () {
+    expect(Tenant::pullPendingTenant())->toBeNull();
+
+    Tenant::createPending();
+
+    expect(Tenant::pullPendingTenant(true))->toBeInstanceOf(Tenant::class);
+});
+
+test('tenancy can create if none are pending', function () {
```

Comment:
I'm a bit confused by this test name @stancl. Seems like the test checks that when we attempt to pull a pending tenant, and there isn't any, a tenant gets automatically created. Does that seem right?

Reply:
Yeah, the test body seems to assert that the method for pulling a tenant creates a tenant if the pending tenant pool is empty.

Reply:
I updated the name

https://github.com/archtechx/tenancy/pull/869/commits/18d12f3a43f2ed9609bafc13f658fe41fc28e8a7